gateway/mw_graphql.go

```diff
@@ -235,3 +264,43 @@ func (p postReceiveHttpHook) Execute(ctx datasource.HookContext, resp *http.Resp
 			},
 		).Debugf("%s.%s: postReceiveHttpHook executed", ctx.TypeName, ctx.FieldName)
 }
+
+type beforeFetchHookV2 struct {
```

Comment:
Why not `GraphQLMiddleware` implements `OnBeforeFetch`, `OnData `...

Reply:
Good idea! Changed!